api/src/soc_mgr.rs

```diff
@@ -0,0 +1,63 @@
+// Licensed under the Apache-2.0 license
+use ureg::MmioMut;
+
```

Comment:
It would be good to add doc comments for the trait and for each field and method that people need to overwrite, since this is a public API to be implemented.

Reply:
I am doing this next in part-2 of caliptra-api.